app/js/views/services-list-item.jsx

```diff
@@ -5,24 +5,57 @@ export default class ServicesListItem extends React.Component {
     super(props);
 
     this.props = props;
-    this.state = props.state;
+    this.state = {
+      // @todo Query getter to get the `available` state.
+      available: true,
+      // @todo Query getter to get the `power` state.
+      on: true,
+    };
 
     this.foxbox = props.foxbox;
   }
 
-  componentWillReceiveProps(props) {
-    this.setState(props.state);
-  }
-
   handleLightOnChange(evt) {
-    // No optimistic update because some network issues can't be caught.
     const on = evt.target.checked;
+    const value = on ? 'On' : 'Off';
+    const id = this.getOperationByAlias('power').id;
+
+    // Optimistic update.
+    this.setState({ on });
+
+    this.foxbox.performSetOperation({
```

Comment:
__question:__ mmm I don't get why can't you just directly pass here operation returned from ```getOperationByAlias```?